circle.yml

```diff
@@ -0,0 +1,16 @@
+dependencies:
+  pre:
+    - go get github.com/axw/gocov/gocov; go get github.com/matm/gocov-html; go get -u github.com/jstemmer/go-junit-report
+test:
+  pre:
+    - go get github.com/mattn/goveralls
+  override:
+    - mkdir -p $CIRCLE_TEST_REPORTS/golang
+    - go test -race -v ./... | go-junit-report > $CIRCLE_TEST_REPORTS/golang/junit.xml
+    - go test -v -cover -race -coverprofile=$CIRCLE_ARTIFACTS/coverage.out ./...
+    - cd $CIRCLE_ARTIFACTS && sed -i '1d' *.out
+    - |
```

Comment:
I have to admit I used the template: https://sites.google.com/a/ft.com/universal-publishing/how-can-i/adding-code-coverage-metrics-to-a-project